delphi-frontend/src/main/java/au/com/integradev/delphi/antlr/ast/node/TextLiteralNodeImpl.java

```diff
@@ -167,26 +170,38 @@ private String createSingleLineValue() {
     return imageBuilder.toString();
   }
 
-  private static char characterEscapeToChar(String image) {
+  private boolean isHighCharUnicode() {
+    return getAst()
+        .getDelphiFile()
+        .getCompilerSwitchRegistry()
+        .isActiveSwitch(SwitchKind.HIGHCHARUNICODE, getTokenIndex());
+  }
+
+  public Charset getAnsiCharset() {
+    return Charset.forName(System.getProperty("native.encoding"));
```

Comment:
Assuming that the compiler is actually using the configured codepage to interpret these escapes, we should:
- read `DCC_Codepage` from dproj files
- emit a warning if conflicting `DCC_Codepage` values are found
- expose an analyzer property to override `DCC_Codepage` and ignore it altogether
- fall back to the system's native encoding if none of these are provided